tcp.rflx

```diff
@@ -0,0 +1,124 @@
+
+    type Kind is
+       (End_Of_Options_List    => 16#00#,
+        No_Operation           => 16#01#,
+        Maximum_Segment_Size   => 16#02#,
+        Window_Scale           => 16#03#,
+        SACK_Permitted         => 16#04#,
+        SACK                   => 16#05#,
+        Timestamp              => 16#08#)
+    with Size => 8;
+
+    type Length is range 2 .. 2**8 - 1 with Size => 8;
+    type TS_Length is range 10 .. 10 with Size => 8;
+    type TSval is mod 2**32;
+    type TSecr is mod 2**32;
+    type Shift_cnt is mod 2**8;
+    type MaxSegSize is mod 2**16;
+
+
+    type Option is
+       message
+          Kind : Kind
+             then null
+                if Kind = End_Of_Options_List or Kind = No_Operation
+             then TS_Length
+                if Kind = Timestamp
+             then Length
+                if Kind = Maximum_Segment_Size
+                   or Kind = Window_Scale
+                   or Kind = SACK_Permitted
+                   or Kind = SACK;
+          TS_Length : TS_Length;
+          TSval : TSval
+                if Kind = Timestamp;
+          TSecr : TSecr
+                if Kind = Timestamp
+             then null
+                if Kind = Timestamp;
+          Length : Length
+             then Shift_cnt
+                if Kind = Window_Scale
+             then MaxSegSize
+                if Kind = Maximum_Segment_Size
+             then SACK_Data
+                if Kind = SACK
+             then null
+                if Kind = SACK_Permitted;
+          Shift_cnt : Shift_cnt
+             then null
+                if Kind = Window_Scale;
+          MaxSegSize : MaxSegSize
+             then null
+                if Kind = Maximum_Segment_Size;
+          SACK_Data : Opaque
+              with Size => Length * 8 - Kind'Size - Length'Size
+             then null
+                if Kind = SACK;
+       end message;
+
+    type Options is array of Option;
+
+    type Segment is
+        message
+            Source_Port : Source_Port;
+            Destination_Port : Destination_Port;
+            Sequence_Number : Sequence_Number;
+            Acknowledgment_Number : Acknowledgment_Number;
+            Data_Offset : Data_Offset;
+            Reserved : Reserved;
+            -- NS, CWR and ECN are experimental, added by RFC 3168 and 3540
+            NS : Boolean;
+            CWR : Boolean;
+            ECN : Boolean;
+
+            -- The following flags are defined in RFC 793
+
+            -- Urgent Pointer field
+            URG : Boolean;
+            -- Acknowledgement field
+            ACK : Boolean;
+            -- Push function
+            PSH : Boolean;
+            -- Reset the connection
+            RST : Boolean;
+            -- Synchronize sequence numbers
+            SYN : Boolean;
+            -- No more data from sender
+            FIN : Boolean;
+
+            Window : Window;
+            Checksum : Checksum;
+            Urgent_Pointer: Urgent_Pointer;
```

Comment:
Missing space before `:`.

Reply:
Changed in dc9f40f3b2fde6d4029438a000055e9658205250.